examples/modals/uploader.py

```diff
@@ -0,0 +1,111 @@
+from typing import Optional
+import discord
+from discord import app_commands
+
+import traceback
+
+# The guild in which this slash command will be registered.
+# It is recommended to have a test guild to separate from your "production" bot
+TEST_GUILD = discord.Object(0)
+# The user ID of the bot owner. This is used to restrict access to the command.
+BOT_OWNER_ID = 0  # Replace with your user ID
+
+
+class MyClient(discord.Client):
+    # Suppress error on the User attribute being None since it fills up later
+    user: discord.ClientUser
+
+    def __init__(self) -> None:
+        # Just default intents and a `discord.Client` instance
+        # We don't need a `commands.Bot` instance because we are not
+        # creating text-based commands.
+        intents = discord.Intents.default()
+        super().__init__(intents=intents)
+
+        # We need an `discord.app_commands.CommandTree` instance
+        # to register application commands (slash commands in this case)
+        self.tree = app_commands.CommandTree(self)
+
+    async def on_ready(self):
+        print(f'Logged in as {self.user} (ID: {self.user.id})')
+        print('------')
+
+    async def setup_hook(self) -> None:
+        # Sync the application command with Discord.
+        await self.tree.sync(guild=TEST_GUILD)
```

Comment:
Again, practicing our preaching. We don't sync on startup and we advise everyone not to.

I'm not sure if this is what we do in other examples, but if so, a comment explaining to not do this in "real" bots would be ideal.